PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['account.journal']
-    _inherit = ['account.move']
-    _inherit = ['ir.actions.report']
-    _inherit = ['ir.attachment']

Total Changes: 12

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_edi/models/account_edi_document.py
+++ b/addons/account_edi/models/account_edi_document.py
@@ -15,6 +15,7 @@ DEFAULT_BLOCKING_LEVEL = 'error'
 
 
 class AccountEdiDocument(models.Model):
+    _name = 'account.edi.document'
     _description = 'Electronic Document for an account.move'
 
     # == Stored fields ==

--- a/addons/account_edi/models/account_edi_format.py
+++ b/addons/account_edi/models/account_edi_format.py
@@ -19,6 +19,7 @@ _logger = logging.getLogger(__name__)
 
 
 class AccountEdiFormat(models.Model):
+    _name = 'account.edi.format'
     _description = 'EDI format'
 
     name = fields.Char()

--- a/addons/account_edi/models/account_journal.py
+++ b/addons/account_edi/models/account_journal.py
@@ -8,7 +8,7 @@ from collections import defaultdict
 
 
 class AccountJournal(models.Model):
-    _inherit = ['account.journal']
+    _inherit = 'account.journal'
 
     edi_format_ids = fields.Many2many(comodel_name='account.edi.format',
                                       string='Electronic invoicing',

--- a/addons/account_edi/models/account_move.py
+++ b/addons/account_edi/models/account_move.py
@@ -10,7 +10,7 @@ from odoo.exceptions import UserError
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     edi_document_ids = fields.One2many(
         comodel_name='account.edi.document',

--- a/addons/account_edi/models/account_move_send.py
+++ b/addons/account_edi/models/account_move_send.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class AccountMoveSend(models.AbstractModel):
-    _inherit = ['account.move.send']
+    _inherit = 'account.move.send'
 
     @api.model
     def _get_mail_attachment_from_doc(self, doc):

--- a/addons/account_edi/models/ir_actions_report.py
+++ b/addons/account_edi/models/ir_actions_report.py
@@ -7,7 +7,7 @@ from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
 
 
 class IrActionsReport(models.Model):
-    _inherit = ['ir.actions.report']
+    _inherit = 'ir.actions.report'
 
     def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
         # EXTENDS base

--- a/addons/account_edi/models/ir_attachment.py
+++ b/addons/account_edi/models/ir_attachment.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class IrAttachment(models.Model):
-    _inherit = ['ir.attachment']
+    _inherit = 'ir.attachment'
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_government_document(self):
